### DISCUSSION TITLE
Trying to understand why one LinearRing does not contain another

### DISCUSSION THREAD

**Discussion by jrnorth** [locationtech/jts#869] [Q&A] [answered]
Hi all,

I've created a small sample repository that demonstrates my problem: https://github.com/jrnorth/jts-test

I have one 'exterior' LinearRing and one 'interior' LinearRing, and I am trying to understand why JTS says the exterior does not contain the interior (using the `contains()` method).

I've been using https://arthur-e.github.io/Wicket/sandbox-gmaps3.html to plot my rings on a map, and all the points of the interior are inside the exterior.

![Screen Shot 2022-05-12 at 6 00 11 PM](https://user-images.githubusercontent.com/4495447/168190780-4560bab6-f9df-44fd-ba1a-c79e39364102.jpg)

**Comment by dr-jts** [comment, marked as answer]
`LinearRing`s are lines, not areas.  So their interior consists of the points in the line, but does not include the area they enclose.

If you turn the exterior ring into a `Polygon` then the `contains()` test will work as you are expecting.

**Reply by jrnorth** [reply]
Thanks. So when would `contains()` return true for two `LinearRing`s? Would they have to have exactly the same points?

**Reply by dr-jts** [reply]
> Thanks. So when would `contains()` return true for two `LinearRing`s? Would they have to have exactly the same points?

Yes, or if one ring is a subset of the other (topologically).
